shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -574,6 +580,13 @@ - (NSObject*)valuePublishedByPlugin:(NSString*)pluginKey {
   return _pluginPublications[pluginKey];
 }
 
+#pragma mark - Memory Notifications
+
+- (void)onMemoryWarning:(NSNotification*)notification {
+  self.shell.PurgeCaches();
```

Comment:
I believe the shell only exists when the `runWithFoo:` method is called. The memory notification may be delivered before that point (presumably). This needs a `nullptr` check on the shell. The channel is fine because it is a message to `nil`.

Reply:
Good catch thanks